src/DataProtection/DataProtection/test/AuthenticatedEncryption/ConfigurationModel/AuthenticatedEncryptorDescriptorTests.cs

```diff
@@ -76,6 +76,31 @@ public void CreateAuthenticatedEncryptor_RoundTripsData_CngGcmImplementation(Enc
             Assert.Equal(plaintext, roundTripPlaintext);
         }
 
+        [ConditionalTheory]
+        [ConditionalRunTestOnlyOnWindows]
+        [InlineData(EncryptionAlgorithm.AES_128_GCM)]
+        [InlineData(EncryptionAlgorithm.AES_192_GCM)]
+        [InlineData(EncryptionAlgorithm.AES_256_GCM)]
+        public void CreateAuthenticatedEncryptor_RoundTripsData_AesGcmImplementation(EncryptionAlgorithm encryptionAlgorithm)
+        {
+            // Parse test input
+            int keyLengthInBits = Int32.Parse(Regex.Match(encryptionAlgorithm.ToString(), @"^AES_(?<keyLength>\d{3})_GCM$").Groups["keyLength"].Value, CultureInfo.InvariantCulture);
+
+            // Arrange
+            var masterKey = Secret.Random(512 / 8);
+            var control = new AesGcmAuthenticatedEncryptor(
```

Comment:
The _control_ should be an instance of `CngGcmAuthenticatedEncryptor` (see line 65 above for an example). That way the control uses CNG, and the test uses the managed `AesGcm` type. As it's written currently, both the control and the test use the managed `AesGcm` type.